### PR TITLE
[sdk] update to GSDK 4.4.3

### PR DESCRIPTION
**Pull request by lmnotran**

### CODE REVIEW

File: src/src/alarm.c [hunk 1]
```diff
@@ -50,31 +50,90 @@
...
+        otSysEventSignalPending();
+    }
+    else
+    {
+        RAIL_Status_t      status;
+        wrap_timer_data_t *timer_data = (wrap_timer_data_t *)cbArg;
 
-    otSysEventSignalPending();
+        if (timer_data->overflow_counter < timer_data->overflow_max)
+        {
```
Comment (hemanth-silabs) [inline src/src/alarm.c:123]: Not for this PR, but I think we can combine the two timer expired callbacks.

**Review by hemanth-silabs — APPROVED** [review]
LGTM 👍

File: src/src/alarm.c (hunk 1 above)
Comment (bertoldi-silabs) [reply src/src/alarm.c:123]: I agree, but I also have that as an action item for myself with planned alarm updates for renode support, so please don't duplicate any work!

**Review by romacdon — APPROVED** [review]
LGTM

**Review by parag-silabs — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**